HLTrigger/HLTfilters/plugins/HLT2L1TkMuonL1TkMuonMuRefDR.cc

```diff
@@ -0,0 +1,192 @@
+
+  descriptions.add("HLT2L1TkMuonL1TkMuonMuRefDR", desc);
+}
+
+bool HLT2L1TkMuonL1TkMuonMuRefDR::getCollections(edm::Event& iEvent,
+                                                 std::vector<l1t::TkMuonRef>& coll1,
+                                                 std::vector<l1t::TkMuonRef>& coll2,
+                                                 trigger::TriggerFilterObjectWithRefs& filterproduct) const {
+  edm::Handle<trigger::TriggerFilterObjectWithRefs> handle1, handle2;
+  if (iEvent.getByToken(inputToken1_, handle1) and iEvent.getByToken(inputToken2_, handle2)) {
+    // get hold of pre-filtered object collections
+    handle1->getObjects(trigger::TriggerObjectType::TriggerL1TkMu, coll1);
+    handle2->getObjects(trigger::TriggerObjectType::TriggerL1TkMu, coll2);
+    const trigger::size_type n1(coll1.size());
+    const trigger::size_type n2(coll2.size());
+
+    if (saveTags()) {
+      edm::InputTag tagOld;
+      for (unsigned int i = 0; i < originTag1_.size(); ++i) {
+        filterproduct.addCollectionTag(originTag1_[i]);
+      }
+      tagOld = edm::InputTag();
+      for (trigger::size_type i1 = 0; i1 != n1; ++i1) {
+        const edm::ProductID pid(coll1[i1].id());
+        const std::string& label(iEvent.getProvenance(pid).moduleLabel());
+        const std::string& instance(iEvent.getProvenance(pid).productInstanceName());
+        const std::string& process(iEvent.getProvenance(pid).processName());
+        edm::InputTag tagNew(edm::InputTag(label, instance, process));
+        if (tagOld.encode() != tagNew.encode()) {
+          filterproduct.addCollectionTag(tagNew);
+          tagOld = tagNew;
+        }
+      }
+      for (unsigned int i = 0; i < originTag2_.size(); ++i) {
+        filterproduct.addCollectionTag(originTag2_[i]);
+      }
+      tagOld = edm::InputTag();
+      for (trigger::size_type i2 = 0; i2 != n2; ++i2) {
+        const edm::ProductID pid(coll2[i2].id());
+        const std::string& label(iEvent.getProvenance(pid).moduleLabel());
+        const std::string& instance(iEvent.getProvenance(pid).productInstanceName());
+        const std::string& process(iEvent.getProvenance(pid).processName());
+        edm::InputTag tagNew(edm::InputTag(label, instance, process));
+        if (tagOld.encode() != tagNew.encode()) {
+          filterproduct.addCollectionTag(tagNew);
+          tagOld = tagNew;
+        }
+      }
+    }
+
+    return true;
+  } else
+    return false;
+}
+
+bool HLT2L1TkMuonL1TkMuonMuRefDR::computeDR(edm::Event& iEvent, l1t::TkMuonRef& r1, l1t::TkMuonRef& r2) const {
+  float muRef1_eta = 0.;
+  float muRef1_phi = 0.;
+  if (r1->muonDetector() != 3) {
+    if (r1->muRef().isNull())
+      return false;
+
+    muRef1_eta = r1->muRef()->hwEta() * 0.010875;
+    muRef1_phi = static_cast<float>(l1t::MicroGMTConfiguration::calcGlobalPhi(
+        r1->muRef()->hwPhi(), r1->muRef()->trackFinderType(), r1->muRef()->processor()));
+    muRef1_phi = muRef1_phi * 2. * M_PI / 576.;
```

Comment:
My suggestions would be to declare the same constants in this plugin, and use the constants instead of the bare numbers.

A following step could be to move the constants to a public header file under `L1Trigger/L1TTrackMatch` and reuse them from there - but MO that's beyond the scope of this PR, and should be done by L1T.